File: src/main/java/gr/apt/persistence/enumeration/LeaveType.java
package gr.apt.persistence.enumeration;

import java.util.List;

public enum LeaveType {
    PAID_LEAVE,
    UNPAID_LEAVE,
    SICKNESS_LEAVE,
    MATERNITY_LEAVE,
    MARRIAGE_LEAVE,
    PARENTAL_LEAVE;

    public static List<String> getValuesList(){
        return List.of(
                LeaveType.PAID_LEAVE.toString(),
                LeaveType.UNPAID_LEAVE.toString(),
                LeaveType.SICKNESS_LEAVE.toString(),
                LeaveType.MATERNITY_LEAVE.toString(),
                LeaveType.MARRIAGE_LEAVE.toString(),
                LeaveType.PARENTAL_LEAVE.toString()
        );
    }
}

File: src/main/java/gr/apt/lms/persistence/enumeration/LeaveType.kt
package gr.apt.lms.persistence.enumeration

enum class LeaveType {
    PAID_LEAVE, UNPAID_LEAVE, SICKNESS_LEAVE, MATERNITY_LEAVE, MARRIAGE_LEAVE, PARENTAL_LEAVE;

    companion object {
        @JvmStatic
        val valuesList: List<String>
            get() = java.util.List.of(
                PAID_LEAVE.toString(),
                UNPAID_LEAVE.toString(),
                SICKNESS_LEAVE.toString(),
                MATERNITY_LEAVE.toString(),
                MARRIAGE_LEAVE.toString(),
                PARENTAL_LEAVE.toString()
            )
    }
}